pages/guides/developing/encrypted-vod.en-US.mdx

```diff
@@ -0,0 +1,186 @@
+
+To encrypt a video file with the key we just generated, we follow these steps:
+
+1. Generate a random 128-bit initialization vector (IV).
+2. Pad the data using PKCS#7 padding.
+3. Encrypt the data using AES-CBC.
+
+```go filename="main.go"
+// Read the input file
+data, err := ioutil.ReadFile("your/local/path/source.mp4")
+if err != nil {
+  log.Fatalf("Error reading input file: %v", err)
+}
+
+// Generate random 128-bit IV
+iv := make([]byte, aes.BlockSize)
+_, err = rand.Read(iv)
+if err != nil {
+  log.Fatalf("Error generating random IV: %v", err)
+}
+
+// Pad the data using PKCS#7 padding
+paddedData := pkcs7Pad(data, aes.BlockSize)
+
+// Encrypt the data using AES-CBC with PKCS#7 padding
+block, err := aes.NewCipher(key)
+if err != nil {
+  log.Fatalf("Error creating AES cipher: %v", err)
+}
+ciphertext := make([]byte, len(paddedData))
+mode := cipher.NewCBCEncrypter(block, iv)
+mode.CryptBlocks(ciphertext, paddedData)
+
+// Write the encrypted data and IV to the output file
+encryptedData := make([]byte, aes.BlockSize+len(ciphertext))
+copy(encryptedData[:aes.BlockSize], iv)
+copy(encryptedData[aes.BlockSize:], ciphertext)
+err = ioutil.WriteFile("your/local/path/encrypted.mp4", encryptedData, 0644)
+if err != nil {
+  log.Fatalf("Error writing encrypted data to output file: %v", err)
+}
+```
+<Callout>
+  Currently Livepeer supports only video content encrypted using AES-CBC with PKCS#7 padding.
+</Callout>
+
+## Step 3: Retrieve the Livepeer Public Key
+
+After obtaining an encryption key and encrypting a video file with it, the next step is to retrieve the Livepeer Public Key. This key will be used to encrypt our encryption key, which can then be shared with Livepeer when creating a video asset.
+
+```go
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"log"
+)
+
+type Response struct {
+	PublicKey string `json:"public_key"`
+}
+
+func main() {
+	publicKey, err := fetchPublicKey()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println(publicKey)
+}
+
+func fetchPublicKey() (string, error) {
+	// Create a new request using http
+	req, err := http.NewRequest("GET", "https://livepeer.studio/api/access-control/public-key", nil)
+	if err != nil {
+		return "", err
+	}
+
+	// Add authorization header to the req
+	req.Header.Add("Authorization", "Bearer XXXX-XXXXXXX-XXXXXX-XXXXX")
+
+	// Send req using http Client
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	
+	defer resp.Body.Close()
+
+	// Check the server response for errors
+	if resp.StatusCode != 200 {
+		return "", fmt.Errorf("server returned: %s", resp.Status)
+	}
+
+	// Read the response body
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	// Unmarshal the JSON response
+	var response Response
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return "", err
+	}
+
+	return response.PublicKey, nil
+}
+```
```

Comment:
Same, any Go examples will have a lot of boilerplate since we don't optimize for that devex today. This would be hugely simplified as a JS example using our SDK. Or probably have the "encryption key encryption" handled by our SDK as well, and not even need this step (and the step below) in the guide.